app/controllers/admin/works_controller.rb

```diff
@@ -381,4 +408,36 @@ def cancel_url
       admin_works_path
     end
     helper_method :cancel_url
+
+    def work_audio_members
+      @work.members.to_a.select do |m|
+        (m.is_a? Asset) && m.content_type && m.content_type.start_with?("audio/")
+      end
+    end
+    helper_method :work_audio_members
+
+    def combined_mp3_audio
+      return nil unless @work.genre.present?
+      return nil unless @work.genre.include?('Oral histories')
+      return nil unless work_audio_members.count > 0
+      oh_content = @work.oral_history_content!
+      oh_content.combined_audio_mp3&.url(public:true)
+    end
+    helper_method :combined_mp3_audio
+
+    def combined_webm_audio
+      return nil unless @work.genre.present?
+      return nil unless @work.genre.include?('Oral histories')
+      return nil unless work_audio_members.count > 0
+      oh_content = @work.oral_history_content!
+      oh_content.combined_audio_webm&.url(public:true)
+    end
+    helper_method :combined_webm_audio
+
+    def combined_audio_fingerprint
+      return nil unless @work.genre.present?
+      return nil unless @work.genre.include?('Oral histories')
+      @work.oral_history_content!.combined_audio_fingerprint
+    end
+    helper_method :combined_audio_fingerprint
```

Comment:
These are an awful lot of methods to include in the controller for a very focused/targetted functional area. 

Let's figure out how to extract them into a presenter or ViewModel or something? If these are all for the UI component for the button to trigger creation and show status, maybe it would be a good place for a ViewModel for that purpose, which can include the helper methods?

Let me know if you're not sure what I mean or want to discuss the best way to do it.

Reply:
Yeah, a presenter is a fine idea. Not a huge change either. I can take care of that.